testing/cuda/init_arch-all-deprecated.cmake

```diff
@@ -16,7 +16,6 @@
 include(${rapids-cmake-dir}/cuda/init_architectures.cmake)
 
 
-set(CMAKE_ERROR_DEPRECATED ON)
```

Comment:
Is it worth updating `add_cmake_test` to always set this variable when building all tests? Do we always want to write tests such that deprecations are caught as errors?